web/src/engine/websites/VortexScans.ts

```diff
@@ -0,0 +1,105 @@
+import { Tags } from '../Tags';
+import icon from './VortexScans.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchWindowScript } from '../platform/FetchProvider';
+
+type APIMangas = {
+    posts: APIManga[]
+}
+
+type APIChapters = {
+    post: APIManga
+}
+
+type APIManga = {
+    id: number,
+    slug: string,
+    postTitle: string,
+    chapters: APIChapter[]
+}
+
+type APIChapter = {
+    id: number,
+    slug: string,
+    number: number,
+    title: string
+}
+
+type MangaID = {
+    id: number,
+    slug: string,
+    title?: string
+}
+
+type APIPages = {
+    chapter: {
+        images: {
+            url: string
+        }[]
+    }
+}
+
+const mangaIdScript = `
+    new Promise((resolve, reject) => {
+       const myregexp = /"postId"\\s*:\\s*(\\d+)/i;
+        __next_f.forEach(element => {
+            const el = element[1];
+            if (el) {
+                if((myregexp.test(el)))  {
+                   const id = parseInt(el.match(myregexp)[1]);
+                   const title = document.querySelector('meta[property="og:title"]').content.trim();
+                   const slug = window.location.pathname.split('/').pop();
+                   resolve({id: id, title: title, slug: slug});
```

Comment:
A promise should only be resolved/rejected once. Consider using a for loop instead of `.forEach` and `return resolve(...)`.

Reply:
Or utilize `__next_f.find(el => condition1 && condition2 %% ...)`